stl/inc/ranges

```diff
@@ -6837,6 +6837,602 @@ namespace ranges {
+            const auto _Temp = *this;
+            --(*this);
+
+            return _Temp;
+        }
+
+        constexpr _Iterator& operator+=(difference_type _X) noexcept(
+            (noexcept(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>() +=
+                      iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>{_X})
+                && ...)) // strengthened
+            requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            _Zip::_Tuple_for_each(
+                [&_X]<typename _IteratorType>(_IteratorType& _Itr) { _Itr += iter_difference_t<_IteratorType>{_X}; },
+                _M_Current);
+
+            return *this;
+        }
+
+        constexpr _Iterator& operator-=(difference_type _X) noexcept(
+            (noexcept(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>() -=
+                      iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>{_X})
+                && ...)) // strengthened
+            requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            _Zip::_Tuple_for_each(
+                [&_X]<typename _IteratorType>(_IteratorType& _Itr) { _Itr -= iter_difference_t<_IteratorType>{_X}; },
+                _M_Current);
+
+            return *this;
+        }
+
+        _NODISCARD constexpr auto operator[](difference_type _X) const
+            noexcept((noexcept((_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>())
+                              [iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>{_X}])
+                      && ...)) // strengthened
+            requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            return _Zip::_Tuple_transform(
+                [&_X]<typename _IteratorType>(
+                    _IteratorType& _Itr) -> decltype(auto) { return _Itr[iter_difference_t<_IteratorType>{_X}]; },
+                _M_Current);
+        }
+
+        _NODISCARD_FRIEND constexpr bool operator==(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+            noexcept(_Lhs._M_Current == _Rhs._M_Current)) // strengthened
+            requires(equality_comparable<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+            if constexpr (_Zip::_All_bidirectional<_IsConst, _ViewTypes...>) {
+                return (_Lhs._M_Current == _Rhs._M_Current);
+            } else {
+                return _Zip::_Evaluate_iterator_sentinel_equality(_Lhs._M_Current, _Rhs._M_Current);
+            }
+        }
+
+        _NODISCARD_FRIEND constexpr auto operator<=>(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+            noexcept(_Lhs._M_Current <=> _Rhs._M_Current)) // strengthened
+            requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            return (_Lhs._M_Current <=> _Rhs._M_Current);
+        }
+
+        _NODISCARD_FRIEND constexpr _Iterator operator+(const _Iterator& _Lhs, difference_type _Rhs) noexcept(

```

Comment:
```suggestion
        _NODISCARD_FRIEND constexpr _Iterator operator+(const _Iterator& _Lhs, const difference_type _Rhs) noexcept(
```
We usually qualify `difference_type` parameters with `const` (they are never modified anyway). Occurs below.

Reply:
I also tend to do that type of qualification in my own code, but I didn't add it because I was copying the function signatures exactly from the C++ Working Draft, which also do not use `const`. I'll add these in a commit.